icu4c/source/i18n/ucal.cpp

```diff
@@ -628,6 +628,27 @@ ucal_getCanonicalTimeZoneID(const char16_t* id, int32_t len,
     return reslen;
 }
 
+U_DRAFT int32_t U_EXPORT2
+ucal_getIanaTimeZoneID(const char16_t* id, int32_t len,
+                        char16_t* result, int32_t resultCapacity, UErrorCode* status)
+{
+    if (status == 0 || U_FAILURE(*status)) {
```

Comment:
Why do we need to check  status == 0 here but not other methods? 
I only see ucal_getCanonicalTimeZoneID in this file perform such check.

Reply:
There are other functions which also checks nullptr, such as _createTimeZone, ucal_getDefaultTimeZone, ucal_getHostTimeZone, and some others. Some of the does this differently. For example:

```
U_CAPI int32_t U_EXPORT2
ucal_getDefaultTimeZone(char16_t* result, int32_t resultCapacity, UErrorCode* ec) {
    int32_t len = 0;
    if (ec != nullptr && U_SUCCESS(*ec)) {
        TimeZone* zone = TimeZone::createDefault();
        if (zone == nullptr) {
            *ec = U_MEMORY_ALLOCATION_ERROR;
        } else {
            UnicodeString id;
            zone->getID(id);
            delete zone;
            len = id.extract(result, resultCapacity, *ec);
        }
    }
    return len;
}
```
I think ICU API should be consistent - maybe just not checking it and let the calling application crash. Anyway, we might discuss about our policy for this next week.

Reply:
Agree about check status == 0, per coding guidelines we should not be doing that: https://unicode-org.github.io/icu/userguide/dev/codingguidelines.html#sample-function-with-error-checking

Reply:
@pedberg-icu Thanks for the link. I see a bunch of other code checking status == 0 or status != 0 along with U_FAILURE/U_SUCCESS. We may look at these code later.

Reply:
Updated the C wrapper skip nullptr checking and directly calls TimeZone::getIanaID(). So sts==0 check is gone.